1-Draft/RFCNNNN-AdditionalTelemetry.md

```diff
@@ -0,0 +1,175 @@
+
+### Other features
+
+- Public PowerBi DashBoard to make the aggregated statistics available to the community.
+  Find a mocked-up PowerBi DashBoard with randomly generated data at the end of the document.
+  It includes the following reports:
+  - Unique user trends
+  - Unique devices trends
+  - Types of user &mdash;based on types of executions taking place
+  - VSCode usage trends
+  - Application trends
+  - Hosted Scenario Trends
+  - PowerShellGet installation trends
+- A cmdlet for providing feedback (Send-PSFeedback):
+  - This cmdlet would take in a string and send it back to the PowerShell team
+  - The cmdlet will also send Platform data (OS/version) and Shell data ($PSVersionTable)
+  - The cmdlet will have an -OpenGitHubIssue switch that will allow the user to open a GitHubIssue
+    in the PowerShell/PowerShell repository
+  - The user will have the option to include the the last error using an -IncludeError switch
+  - If the command is not successful (ex. there is no Web Connection) the cmdlet will return a
+    warning and alternative instructions for support through GitHub
+  - If no location switch is used the cmdlet will default send the feedback to the PowerShell
+    User Voice
+  - Based on usage of this cmdlet there would potential for additional parameters
+- Disabling, and re-enabling telemetry will be available through a new cmdlet (Set-PSTelemetry):
+  - The cmdlet will stop all collection of telemetry for the remainder of that session and stop
+    telemetry from being collected on subsequent starts
+  - The user can request their telemetry is deleted by using a Remove-PSTelemetry cmdlet
+  - In order to have their telemetry deleted the user will be required to provide their System GUID
+    and User GUID
+  - If the cmdlet is unable to send the information the cmdlet will return a warning and alternative
+    instructions for support through GitHub. It will be specified that GUIDs are unique identifiers
+    and should not be shared publicly on GitHub.
+- There will be three options for the telemetry environment variable which can all be set manually
+  or using the Set-PSTelemetry cmdlet
+  - $True indicates all telemetry will be collected
+  - $False indicates no telemetry will be collected
+  - $NonIdentifying indicates that only non-identifying telemetry will be collected i.e. no GUIDs
+    will be collected
+- The existing disabling mechanisms for PowerShell telemetry will remain including the ability to
+  set the telemetry environment variable before the first start of PowerShell
+
+### Future work
+
+A public API for module authors and PowerShell hosted application owners to use the same
+infrastructure to collect telemetry for their module or application.
+Making the API public is a phase two goal of this project.
+We will make the telemetry publicly available once it is internally validated.
+Microsoft will not collect, store, or manage any data externally gathered through this API.

```

Comment:
Could you clarify this statement? It seems there _is_ intent to use the data from the telemetry within the PS Team, at the very least. 😄

Reply:
Good point! I will clarify that if external PowerShell authors want to use the API they will need to provide their own storage for the data.

Reply:
Should clarify that the intent is to provide a simple consistent API for PowerShell authors to collect their own telemetry, but they need to provide a storage account of where it will be sent as the PowerShell team won't cover the cost of storing it for them.  Also things like validating compliance/privacy will be up to the user of the API and not something PowerShell team/Microsoft will cover.